addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -264,10 +265,59 @@ ListRenderer.include({
      */
     getEditableRecordID: function () {
         if (this.currentRow !== null) {
-            return this.state.data[this.currentRow].id;
+            return this.getRecordID(this.currentRow);
         }
         return null;
     },
+    /**
+     * for editable group by mode record could be deep down in state tree
+     * traverse state and get record
+     *
+     * @param {string} [recordID]
+     * @returns {object}
+     */
+    getRecord: function (recordID) {
```

Comment:
move this method to basicRenderer (abstract) + implement it in form and list and keep it empty in kanban (not used...)